go.mod

```diff
@@ -87,3 +87,4 @@ require (
 	google.golang.org/api v0.31.0
 	gopkg.in/yaml.v2 v2.4.0
 )
+replace github.com/minio/minio-go => ../minio-go
```

Comment:
there is a minio-go change needed as well?

Reply:
yes - status code returned for stat for a deletemarker version should be MethodNotAllowed rather than BadRequest

Reply:
that is returned by server what is minio-go doing here?

Reply:
minio-go has a bug where the ErrorResponse struct is not constructed correctly when error code is MethodNotAllowed - this is an issue in replication because we need to determine if a delete marker version exists or not before trying to replicate it and ErrorCode is not populated because of the minio-go bug.

Reply:
where is the minio-go change? @poornas

Reply:
https://github.com/minio/minio-go/pull/1531/files